packages/pluggableWidgets/datagrid-web/src/Datagrid.tsx

```diff
@@ -101,15 +112,29 @@ export default function Datagrid(props: DatagridContainerProps): ReactElement {
                 (renderWrapper, columnIndex) => {
                     const { attribute, filter } = props.columns[columnIndex];
                     const [, filterDispatcher] = customFiltersState[columnIndex];
-                    return attribute
-                        ? renderWrapper(
-                              <FilterContext.Provider value={{ filterDispatcher, attribute }}>
-                                  {filter}
-                              </FilterContext.Provider>
-                          )
-                        : renderWrapper(filter);
+                    const initialFilters = extractFilters(attribute, viewStateFilters.current);
+
+                    if (!attribute) {
+                        return renderWrapper(filter);
+                    }
+
+                    return renderWrapper(
+                        <FilterContext.Provider
+                            value={{
+                                filterDispatcher: prev => {
+                                    setFiltered(true);
+                                    filterDispatcher(prev);
```

Comment:
What is the need for a recursive call here?

Reply:
Is not recursive, just the name of the property is the same..